### PR TITLE
refactor: (rewrote into TypeScript)

### PR DESCRIPTION
**Pull request by nester7x**

### CODE REVIEW

File: src/utils/cookieUtils.tsx [hunk 1]
```diff
@@ -8,18 +8,18 @@ export const setCookie = (cname, value, days) => {
   document.cookie = `${cname}=${value || ''}${expires}; path=/`;
 };
 
-export const getCookie = (cname) => {
+export const getCookie = (cname: string): string | undefined => {
   const nameEQ = `${cname}=`;
   const ca = document.cookie.split(';');
   for (let i = 0; i < ca.length; i += 1) {
```
Comment (mrnobody3) [inline src/utils/cookieUtils.tsx]: I'd change variable names, something more informative

**Review by mrnobody3 — APPROVED** [review]

File: src/components/Button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import BtnStyled from './styles';
+
+type BtnProps = {
+  onClick?: () => void;
+  type?: 'button' | 'submit' | 'reset';
+  text?: string;
+  children: ReactNode;
+  disabled?: boolean;
+};
```
Comment (SerhiienkoRuslan) [inline src/components/Button/index.tsx:10]: type BtnProps = {
  onClick: () => void;
  type: 'button' | 'submit' | 'reset';
  text: string;
  children: ReactNode;
  disabled: boolean;
};

File: src/components/Button/index.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+const Btn: FC<BtnProps> = ({ onClick, type, children, ...rest }) => (
```
Comment (SerhiienkoRuslan) [inline src/components/Button/index.tsx]: const Btn: FC<BtnProps> = ({ onClick, type = 'button', children, disabled = false, ...rest }) => (

File: src/context/authContext.tsx [hunk 4]
```diff
@@ -1,29 +1,55 @@
...
 import { api } from 'utils/apiUtils';
 
-export const AuthContext = createContext();
+type User = {
+  id?: string | undefined;
+  username?: string | undefined;
+  email?: string | undefined;
+  roles: string[];
+  [key: string]: any;
+};
```
Comment (SerhiienkoRuslan) [inline src/context/authContext.tsx:12]: type User = {
  id: string;
  username: string;
  email: string;
  roles: string[];
  [key: string]: any;
};

File: src/context/authContext.tsx [hunk 5]
```diff
@@ -1,29 +1,55 @@
...
+  const [loginState, setLoginState] = useState<LoginState>({
     isLoggedIn: false,
     accessToken: null,
     refreshToken: null,
-    errors: ''
+    errors: '',
   });
 
-  const auth = async (endpoint, loginData) => {
+  const auth = async (endpoint: string, loginData: object) => {
```
Comment (SerhiienkoRuslan) [inline src/context/authContext.tsx]: loginData: object -> loginData: { email: string, password: string.... }

File: src/context/authContext.tsx [hunk 6]
```diff
@@ -96,13 +122,13 @@ export const GlobalAuthProvider = ({ children }) => {
       if (loginState.accessToken && !isUserInfo) {
         const data = await api.get('user', loginState.accessToken);
 
-        let userInfo;
+        let userInfo: object | any;
```
Comment (SerhiienkoRuslan) [inline src/context/authContext.tsx]: let userInfo: object | any;  -> userInfo: User

File: src/context/authContext.tsx [hunk 7]
```diff
@@ -113,17 +139,15 @@ export const GlobalAuthProvider = ({ children }) => {
...
 
-  return (
-    <Provider value={{ user, loginState, auth, logout }}>{children}</Provider>
-  );
+  return <Provider value={{ user, loginState, auth, logout }}>{children}</Provider>;
 };
 
 GlobalAuthProvider.propTypes = {
-  children: PropTypes.element.isRequired
+  children: PropTypes.element.isRequired,
```
Comment (SerhiienkoRuslan) [inline src/context/authContext.tsx]: no needed anymore

**Review by SerhiienkoRuslan — APPROVED** [review]

**Pull request merged**